BitwardenShared/Core/Autofill/Utilities/Fido2DebuggingUtilsTests.swift

```diff
@@ -0,0 +1,92 @@
+#if DEBUG
+import BitwardenSdk
+import Foundation
+import XCTest
+
+@testable import BitwardenShared
+
+class Fido2DebuggingUtilsTests: BitwardenTestCase {
+    // MARK: Tests
+
+    /// `static describeAuthDataFlags(_:)` returns the formatted string successfully for the flags.
+    func test_describeAuthDataFlags() {
+        var authData: [UInt8] = []
+        // add some 32 bytes
+        let firstPart = Data((0 ..< 32).map { _ in 1 })
+        authData.append(contentsOf: firstPart)
+        authData.append(217)
+
+        let result = Fido2DebuggingUtils.describeAuthDataFlags(Data(authData))
+        XCTAssertEqual(result, "Flags: UP: 1 - UV: 0 - BE: 1 - BS: 1 - AD: 1 - ED: 1")
+    }
+
+    /// `static describe(request:)` returns a string with the formatted description of the request
+    /// when optionals are not filled.
+    func test_describe_makeCredentialRequest_withoutOptionals() {
+        let request = MakeCredentialRequest(
+            clientDataHash: Data((0 ..< 32).map { _ in 1 }),
+            rp: PublicKeyCredentialRpEntity(id: "someApp.com", name: nil),
+            user: PublicKeyCredentialUserEntity(
+                id: Data((0 ..< 32).map { _ in 1 }),
+                displayName: "userDisplay",
+                name: "user"
+            ),
+            pubKeyCredParams: [PublicKeyCredentialParameters(ty: "public-key", alg: -7)],
+            excludeList: nil,
+            options: Options(rk: true, uv: .preferred),
+            extensions: nil
+        )
+
+        // swiftlint:disable:next line_length
+        let expectedResult = "ClientDataHash: 0101010101010101010101010101010101010101010101010101010101010101\nRP -> Id: someApp.com \nRP -> Name: nil \nUser -> Id: 0101010101010101010101010101010101010101010101010101010101010101\nUser -> Name: user \nUser -> DisplayName: userDisplay \nPubKeyCredParams: [BitwardenSdk.PublicKeyCredentialParameters(ty: \"public-key\", alg: -7)]\nExcludeList: nil \nOptions -> RK: true \nOptions -> UV: preferred\nExtensions: nil \n"
```

Comment:
🎨 You could format this on multiple lines to avoid the long line:
```suggestion
        let expectedResult = """
            ClientDataHash: 0101010101010101010101010101010101010101010101010101010101010101
            RP -> Id: someApp.com
            RP -> Name: nil
            User -> Id: 0101010101010101010101010101010101010101010101010101010101010101
            User -> Name: user
            User -> DisplayName: userDisplay
            PubKeyCredParams: [BitwardenSdk.PublicKeyCredentialParameters(ty: "public-key", alg: -7)]
            ExcludeList: nil
            Options -> RK: true
            Options -> UV: preferred
            Extensions: nil
            """
```